### DISCUSSION TITLE
Elixir Support

### DISCUSSION THREAD

**Discussion by corck** [crate/crate#10296] [General] [answered]
Elixir (Beam) is for sure one of the best backends you can have to build scalable applications with thousands of simultanous connections, (see Whatsapp). 

We from [Citymonitor](https://www.city-monitor.com) build up on it and as we grow we'll need a scalable DB thats why we are looking for solutions we can hook into our system. 

So IoT, Elixir and Crate seems to be a perfect match, would be awesome to have a elixir Ecto adapter for Crate

**Comment by MarkusH** [comment]
Hi @corck, thanks for bringing this up. CrateDB supports the PostgreSQL wire protocol. Did you happen to take a look into using a PostgreSQL adapter? I'm not familiar with Elixir myself, but would https://github.com/elixir-ecto/postgrex be an option?

**Reply by corck** [reply]
We tried that a while ago, better said, Jodok did, and failed due to some tables missing that ecto's test setup requires.

**Reply by amotl** [reply, marked as answer]
Hi again.

> CrateDB supports the PostgreSQL wire protocol. Did you happen to take a look into using a PostgreSQL adapter?

We've verified [Postgrex](https://hexdocs.pm/postgrex/readme.html) works well in 2025, at least for fundamental conversations.

#### Documentation
[Connect to CrateDB from Elixir applications](https://cratedb.com/docs/guide/connect/elixir/)

#### QA
CI workflow: https://github.com/crate/cratedb-examples/actions/workflows/lang-elixir-postgrex.yml ([source](https://github.com/crate/cratedb-examples/blob/main/.github/workflows/lang-elixir-postgrex.yml))
CI workload: https://github.com/crate/cratedb-examples/tree/main/by-language/elixir-postgrex
